[Build Error] 3rdparty qwt building for ROS-noetic on Arch Linux fails

Thanks for contributing to PlotJuggler. You are great!

So are you for creating PlotJuggler!

# If you ignore these instructions, I will ignore your issue

Please allow me to better understand your problem providing the following information:

- Please use always the master branch or the latest tag, if you can.
Latest tag 2.8.3 is used. Doubt switching to latest head on master will help since this a vendored library issue.

- Check if the problem can be reproduced using the dummy data created by the command line argument "-t" or one of the files in the folder "datasamples".

Not applicable

- If it can't be reproduced with the dummy data, __please__ share the CSV file or the rosbag that can be used to reproduce the problem.

Not applicable

![share the rosbag](https://github.com/facontidavide/PlotJuggler/raw/master/docs/images/show_me_the_rosbag.jpg)

Not applicable

When attempting to build this package with `catkin build`:
I am running into the following error:
```
[  7%] Building CXX object 3rdparty/qwt/src/CMakeFiles/qwt_static.dir/qwt_plot_renderer.cpp.o
In file included from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:13:
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_layout.h:84:51: warning: ‘constexpr QFlags<T>::QFlags(QFlags<T>::Zero) [with Enum = QwtPlotLayout::Option; QFlags<T>::Zero = int QFlags<QwtPlotLayout::Option>::Private::*]’ is deprecated: Use default constructor instead [-Wdeprecated-declarations]
   84 |         const QRectF &plotRect, Options options = 0x00 );
      |                                                   ^~~~
In file included from /usr/include/qt/QtCore/qglobal.h:1302,
                 from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_global.h:13,
                 from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.h:13,
                 from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:10:
/usr/include/qt/QtCore/qflags.h:123:80: note: declared here
  123 | ATED_X("Use default constructor instead") Q_DECL_CONSTEXPR inline QFlags(Zero) noexcept : i(0) {}
      |                                                                   ^~~~~~

/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:94:53: error: return type ‘class QPainterPath’ is incomplete
   94 |     const QWidget* canvas, const QRectF &canvasRect )
      |                                                     ^
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp: In function ‘void qwtCanvasClip(const QWidget*, const QRectF&)’:
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:107:18: error: aggregate ‘QPainterPath clipPath’ has incomplete type and cannot be defined
  107 |     QPainterPath clipPath;
      |                  ^~~~~~~~
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp: In member function ‘void QwtPlotRenderer::renderDocument(QwtPlot*, const QString&, const QString&, const QSizeF&, int)’:
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:312:41: warning: ‘virtual void QPdfWriter::setPageSizeMM(const QSizeF&)’ is deprecated: Use setPageSize(QPageSize(size, QPageSize::Millimeter)) instead [-Wdeprecated-declarations]
  312 |         pdfWriter.setPageSizeMM( sizeMM );
      |                                         ^
In file included from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:71:
/usr/include/qt/QtGui/qpdfwriter.h:98:10: note: declared here
   98 |     void setPageSizeMM(const QSizeF &size) override;
      |          ^~~~~~~~~~~~~
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp: In member function ‘virtual void QwtPlotRenderer::render(QwtPlot*, QPainter*, const QRectF&) const’:
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:507:64: warning: ‘void QWidget::getContentsMargins(int*, int*, int*, int*) const’ is deprecated: use contentsMargins() [-Wdeprecated-declarations]
  507 |         plot->getContentsMargins( &left, &top, &right, &bottom );
      |                                                                ^
In file included from /usr/include/qt/QtWidgets/qframe.h:44,
                 from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot.h:16,
                 from /home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:11:
/usr/include/qt/QtWidgets/qwidget.h:530:10: note: declared here
  530 |     void getContentsMargins(int *left, int *top, int *right, int *bottom) const;
      |          ^~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp: In member function ‘virtual void QwtPlotRenderer::renderCanvas(const QwtPlot*, QPainter*, const QRectF&, const QwtScaleMap*) const’:
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:859:22: error: aggregate ‘QPainterPath clipPath’ has incomplete type and cannot be defined
  859 |         QPainterPath clipPath;
      |                      ^~~~~~~~
/home/acxz/vcs/git/github/ros-noetic-arch/ros-noetic-plotjuggler/src/PlotJuggler-2.8.3/3rdparty/qwt/src/qwt_plot_renderer.cpp:883:22: error: aggregate ‘QPainterPath clipPath’ has incomplete type and cannot be defined
  883 |         QPainterPath clipPath;
      |                      ^~~~~~~~
make[2]: *** [3rdparty/qwt/src/CMakeFiles/qwt_static.dir/build.make:212: 3rdparty/qwt/src/CMakeFiles/qwt_static.dir/qwt_plot_renderer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2507: 3rdparty/qwt/src/CMakeFiles/qwt_static.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

Sys info:
OS: Arch Linux
ROS: Noetic
GCC: 10.1.0

Would there be anyway for me to compile against Arch Linux's packaged qwt, instead of using the one vendored inside of plotjuggler?

Thanks!

--- a/3rdparty/qwt/src/qwt_null_paintdevice.cpp
+++ b/3rdparty/qwt/src/qwt_null_paintdevice.cpp
@@ -9,6 +9,7 @@
 
 #include "qwt_null_paintdevice.h"
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 
 class QwtNullPaintDevice::PrivateData
 {

--- a/3rdparty/qwt/src/qwt_painter.cpp
+++ b/3rdparty/qwt/src/qwt_painter.cpp
@@ -17,6 +17,7 @@
 #include <qframe.h>
 #include <qrect.h>
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qpalette.h>
 #include <qpaintdevice.h>
 #include <qpixmap.h>

--- a/3rdparty/qwt/src/qwt_painter_command.h
+++ b/3rdparty/qwt/src/qwt_painter_command.h
@@ -13,6 +13,7 @@
 #include "qwt_global.h"
 
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 #include <qpixmap.h>
 #include <qimage.h>
 #include <qpolygon.h>

--- a/3rdparty/qwt/src/qwt_picker.cpp
+++ b/3rdparty/qwt/src/qwt_picker.cpp
@@ -16,6 +16,7 @@
 
 #include <qevent.h>
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qcursor.h>
 #include <qpointer.h>
 

--- a/3rdparty/qwt/src/qwt_plot_canvas.cpp
+++ b/3rdparty/qwt/src/qwt_plot_canvas.cpp
@@ -14,6 +14,7 @@
 #include "qwt_plot.h"
 
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qstyle.h>
 #include <qstyleoption.h>
 #include <qpaintengine.h>

--- a/3rdparty/qwt/src/qwt_plot_curve.cpp
+++ b/3rdparty/qwt/src/qwt_plot_curve.cpp
@@ -21,6 +21,7 @@
 #include "qwt_graphic.h"
 
 #include <qpainter.h>
+#include <qpainterpath.h>
 
 static inline QRectF qwtIntersectedClipRect( const QRectF &rect, QPainter *painter )
 {

--- a/3rdparty/qwt/src/qwt_plot_panner.cpp
+++ b/3rdparty/qwt/src/qwt_plot_panner.cpp
@@ -17,6 +17,7 @@
 #include <qstyle.h>
 #include <qstyleoption.h>
 #include <qpainter.h>
+#include <qpainterpath.h>
 
 static QBitmap qwtBorderMask( const QWidget *canvas, const QSize &size )
 {

--- a/3rdparty/qwt/src/qwt_plot_renderer.cpp
+++ b/3rdparty/qwt/src/qwt_plot_renderer.cpp
@@ -20,6 +20,7 @@
 #include "qwt_math.h"
 
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qtransform.h>
 #include <qprinter.h>
 #include <qfiledialog.h>

--- a/3rdparty/qwt/src/qwt_widget_overlay.cpp
+++ b/3rdparty/qwt/src/qwt_widget_overlay.cpp
@@ -11,6 +11,7 @@
 #include "qwt_painter.h"
 
 #include <qpainter.h>
+#include <qpainterpath.h>
 #include <qpaintengine.h>
 #include <qimage.h>
 #include <qevent.h>

